src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/DbContextEditorServices.cs

```diff
@@ -122,18 +135,21 @@ public EditSyntaxTreeResult EditStartupForNewContext(ModelType startUp, string d
                     .FirstOrDefault(n => n is MethodDeclarationSyntax
                         && ((MethodDeclarationSyntax)n).Identifier.ToString() == "ConfigureServices") as MethodDeclarationSyntax;
 
-                if (configServicesMethod != null)
+                var configRootProperty = TryGetIConfigurationRootProperty(startUp.TypeSymbol);
+
+                if (configServicesMethod != null && configRootProperty != null)
                 {
                     var servicesParam = configServicesMethod.ParameterList.Parameters
                         .FirstOrDefault(p => p.Type.ToString() == "IServiceCollection") as ParameterSyntax;
 
                     if (servicesParam != null)
                     {
+                        AddConnectionString(dbContextTypeName, dataBaseName);
```

Comment:
Yes, in case the configRootProperty is null, it throws an error saying that it needs that property. 
If the property isn't defined, we could potentially fall back to old way of adding the connection string in the Startup.cs, but that means the user will manually have to do that later on anyways.  

---
In reply to: [66547326](https://github.com/aspnet/Scaffolding/pull/236#discussion_r66547326) [](ancestors = 66547326)